Polygon_mesh_processing/test/Polygon_mesh_processing/connected_component_surface_mesh.cpp

```diff
@@ -130,9 +141,56 @@ int main(int argc, char* argv[])
     CGAL::make_triangle(p,q,r,m);
     CGAL::make_tetrahedron(p,q,r,s,m);
     PMP::keep_large_connected_components(m, 4);
-    m.collect_garbage();
-    assert( num_vertices(m) == 8);
+
+    assert(vertices(m).size() == 8);
   }
+}
+
+void test_CC_with_area_size_map(Mesh sm)
+{
+  std::cout << " -- test with area size map -- " << std::endl;
+
+  typedef boost::graph_traits<Mesh>::face_descriptor                      face_descriptor;
+
+  Kernel k;
+  Face_descriptor_area_functor<Mesh, Kernel> f(sm, k);
+
+  std::cout << "We keep the " << 2 << " largest components" << std::endl;
+  PMP::keep_largest_connected_components(sm, 2,
+                                         PMP::parameters::face_size_map(
+                                           CGAL::internal::boost_::make_function_property_map<face_descriptor>(f)));
+  assert(vertices(sm).size() == 1459);
+
+  {
+    Mesh m;
+
+    Point p(0,0,0), q(1,0,0), r(0,1,0), s(0,0,1);
+    CGAL::make_tetrahedron(p,q,r,s,m);
+    CGAL::make_tetrahedron(p,q,r,s,m);
```

Comment:
did you add twice the same on purpose?

Reply:
It was done like that before to have more connected components. I don't care for it so I just left it how it was.